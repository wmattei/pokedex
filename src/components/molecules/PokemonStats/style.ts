import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    marginTop: 30,
  },
  title: {
    paddingLeft: 15,
    fontSize: 25,
  },
  row: {
    paddingTop: 10,
    justifyContent: 'space-around',
    flexDirection: 'row',
  },
});
