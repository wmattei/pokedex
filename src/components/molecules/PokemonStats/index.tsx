import * as React from 'react';
import { Text, View } from 'react-native';
import { IPokemonStats } from '../../../types/pokemon';
import StatCard from '../../atoms/StatCard';
import style from './style';

type Props = {
  stats: IPokemonStats;
};

export default function PokemonStats({ stats }: Props) {
  return (
    <View style={style.container}>
      <Text style={style.title}>Stats:</Text>
      <View style={style.row}>
        {stats.hp && <StatCard label={'HP'} value={stats.hp} />}
        {stats.attack && <StatCard label={'Ataque'} value={stats.attack} />}
        {stats.defense && <StatCard label={'Defesa'} value={stats.defense} />}
      </View>
      <View style={style.row}>
        {stats.specialAttack && (
          <StatCard label={'Ataque especial'} value={stats.specialAttack} />
        )}
        {stats.specialDefense && (
          <StatCard label={'Defesa especial'} value={stats.specialDefense} />
        )}
        {stats.speed && <StatCard label={'Velocidade'} value={stats.speed} />}
      </View>
    </View>
  );
}
