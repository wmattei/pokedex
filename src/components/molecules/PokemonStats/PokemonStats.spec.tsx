import * as React from 'react';
import { create } from 'react-test-renderer';
import PokemonStats from '.';

describe('PokemonsStats', () => {
  it('Renders correctly', () => {
    const stats = {
      attack: 1,
      defense: 1,
      hp: 1,
      specialAttack: 1,
      specialDefense: 1,
      speed: 1,
    };
    const tree = create(<PokemonStats stats={stats} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
