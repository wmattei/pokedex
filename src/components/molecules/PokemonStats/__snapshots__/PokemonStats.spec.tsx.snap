// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonsStats Renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "paddingLeft": 15,
      }
    }
  >
    Stats:
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6200ee",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "space-evenly",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6200ee",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
      <Text>
        HP
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6200ee",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "space-evenly",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6200ee",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
      <Text>
        Ataque
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6200ee",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "space-evenly",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6200ee",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
      <Text>
        Defesa
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6200ee",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "space-evenly",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6200ee",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
      <Text>
        Ataque especial
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6200ee",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "space-evenly",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6200ee",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
      <Text>
        Defesa especial
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#6200ee",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "space-evenly",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6200ee",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
      <Text>
        Velocidade
      </Text>
    </View>
  </View>
</View>
`;
